openmdao/drivers/doe_driver.py

```diff
@@ -0,0 +1,262 @@
+"""
+Design-of-Experiments Driver.
+"""
+from __future__ import print_function
+
+import sys
+import traceback
+import inspect
+
+import numpy as np
+
+from openmdao.core.driver import Driver, RecordingDebugging
+from openmdao.core.analysis_error import AnalysisError
+
+from openmdao.utils.mpi import MPI
+
+
+class DOEGenerator(object):
+    """
+    Base class for a callable object that generates cases for a DOEDriver.
+
+    Attributes
+    ----------
+    _num_samples : int
+        The number of samples generated (available after generator has been called).
+    """
+
+    def __init__(self):
+        """
+        Initialize the DOEGenerator.
+        """
+        self._num_samples = 0
+
+    def __call__(self, design_vars):
+        """
+        Generate case.
+
+        Parameters
+        ----------
+        design_vars : dict
+            Dictionary of design variables for which to generate values.
+
+        Yields
+        ------
+        list
+            list of name, value tuples for the design variables.
+        """
+        return
```

Comment:
return followed by yield

Reply:
This is intentional.. I found this to be the cleanest way to implement an empty iterator that works fro both py2 and py3.  Open to better suggestions.

Reply:
could you just return an empty tuple for example, or, if it has to be an iterator, return iter(())?